DigitalLearningSolutions.Data/DataServices/NotificationDataService.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace DigitalLearningSolutions.Data.DataServices
+{
+    using System.Data;
+    using System.Linq;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models;
+
+    public interface INotificationDataService
+    {
+        UnlockData? GetUnlockData(int progressId);
+
+        ProgressCompletionData? GetProgressCompletionData(int progressId, int candidateId, int customisationId);
+    }
+
+    public class NotificationDataService : INotificationDataService
+    {
+        private readonly IDbConnection connection;
+
+        public NotificationDataService(IDbConnection connection)
+        {
+            this.connection = connection;
+        }
+
+        public UnlockData? GetUnlockData(int progressId)
+        {
+            return connection.Query<UnlockData?>(
+                @"SELECT TOP (1)
+                        candidates.EmailAddress AS DelegateEmail,
+                        candidates.FirstName + ' ' + candidates.LastName AS DelegateName,
+                        centres.ContactForename,
+                        COALESCE (centres.NotifyEmail, COALESCE (centres.ContactEmail, centres.pwEmail)) AS ContactEmail,
+                        applications.ApplicationName + ' - ' + customisations.CustomisationName AS CourseName,
+                        progress.CustomisationID AS CustomisationId
+                    FROM Progress AS progress
+                        INNER JOIN Candidates AS candidates
+                            ON progress.CandidateID = candidates.CandidateID
+                        INNER JOIN Centres AS centres
+                            ON candidates.CentreID = centres.CentreID
+                        INNER JOIN Customisations AS customisations
+                            ON progress.CustomisationID = customisations.CustomisationID
+                        INNER JOIN Applications AS applications
+                            ON customisations.ApplicationID = applications.ApplicationID
+                    WHERE  (progress.ProgressID = @progressID)",
+                new { progressId }
+            ).FirstOrDefault();
+        }
+
+        public ProgressCompletionData? GetProgressCompletionData(int progressId, int candidateId, int customisationId)

```

Comment:
Why is this in this data service? It doesn't seem to have anything to do with it?

Reply:
Looking at the existing method in this file, it seemed to me like this Data Service was for getting info from a variety of tables for the purpose of sending an email. Should this be in the ProgressDataService instead perhaps?